### ISSUE TITLE
TypeError using `expand` 

### ISSUE THREAD

**Issue by augustinas1** [JuliaSymbolics/SymbolicUtils.jl#330]
```julia
@syms a b c d
expand(a*b + a*b*c*d + a)
```
```julia
ERROR: TypeError: non-boolean (SymbolicUtils.Term{Bool, Nothing}) used in boolean context
Stacktrace:
  [1] _remove_zeros!(p::DynamicPolynomials.Polynomial{true, SymbolicUtils.Mul{Number, Int64, Dict{Any, Number}, Nothing}})
    @ DynamicPolynomials C:\Users\asukys\.julia\packages\DynamicPolynomials\mpulO\src\poly.jl:148
  [2] Polynomial
    @ C:\Users\asukys\.julia\packages\DynamicPolynomials\mpulO\src\poly.jl:14 [inlined]
  [3] Polynomial
    @ C:\Users\asukys\.julia\packages\DynamicPolynomials\mpulO\src\poly.jl:30 [inlined]
  [4] DynamicPolynomials.Polynomial(t::DynamicPolynomials.Term{true, SymbolicUtils.Mul{Number, Int64, Dict{Any, Number}, Nothing}})
    @ DynamicPolynomials C:\Users\asukys\.julia\packages\DynamicPolynomials\mpulO\src\poly.jl:60
  [5] convertconstant
    @ C:\Users\asukys\.julia\packages\DynamicPolynomials\mpulO\src\poly.jl:53 [inlined]
  [6] convert
    @ C:\Users\asukys\.julia\packages\MultivariatePolynomials\vqcb5\src\conversion.jl:4 [inlined]
  [7] setindex!(A::Vector{DynamicPolynomials.Polynomial{true, Any}}, x::SymbolicUtils.Mul{Number, Int64, Dict{Any, Number}, Nothing}, i1::Int64)
    @ Base .\array.jl:839
  [8] copyto!(dest::Vector{DynamicPolynomials.Polynomial{true, Any}}, src::Tuple{DynamicPolynomials.Polynomial{true, Int64}, SymbolicUtils.Mul{Number, Int64, Dict{Any, Number}, Nothing}})
    @ Base .\abstractarray.jl:846
  [9] vect(::DynamicPolynomials.Polynomial{true, Int64}, ::Vararg{Any, N} where N)
    @ Base .\array.jl:129
 [10] (::SymbolicUtils.ACRule{typeof(Combinatorics.permutations), SymbolicUtils.Rule{SymbolicUtils.Term{Any, Nothing}, SymbolicUtils.var"#term_matcher#102"{Tuple{SymbolicUtils.var"#literal_matcher#98"{typeof(+)}, SymbolicUtils.var"#slot_matcher#99"{SymbolicUtils.Slot{typeof(SymbolicUtils.ismpoly)}}, SymbolicUtils.var"#slot_matcher#99"{SymbolicUtils.Slot{typeof(SymbolicUtils.ismpoly)}}}}, SymbolicUtils.var"#117#129"}})(term::SymbolicUtils.Add{Number, Int64, Dict{Any, Number}, Nothing})
    @ SymbolicUtils C:\Users\asukys\.julia\packages\SymbolicUtils\2UXNG\src\rule.jl:332
 [11] (::SymbolicUtils.Rewriters.RestartedChain{Vector{SymbolicUtils.AbstractRule}})(x::SymbolicUtils.Add{Number, Int64, Dict{Any, Number}, Nothing})
    @ SymbolicUtils.Rewriters C:\Users\asukys\.julia\packages\SymbolicUtils\2UXNG\src\rewriters.jl:77
 [12] (::SymbolicUtils.Rewriters.PassThrough{SymbolicUtils.Rewriters.RestartedChain{Vector{SymbolicUtils.AbstractRule}}})(x::SymbolicUtils.Add{Number, Int64, Dict{Any, Number}, Nothing})
    @ SymbolicUtils.Rewriters C:\Users\asukys\.julia\packages\SymbolicUtils\2UXNG\src\rewriters.jl:132
 [13] (::SymbolicUtils.Rewriters.Walk{:post, SymbolicUtils.Rewriters.PassThrough{SymbolicUtils.Rewriters.RestartedChain{Vector{SymbolicUtils.AbstractRule}}}, SymbolicUtils.var"#simterm#125"{SymbolicUtils.var"#simterm#114#126"}, false})(x::SymbolicUtils.Add{Number, Int64, Dict{Any, Number}, Nothing})
    @ SymbolicUtils.Rewriters C:\Users\asukys\.julia\packages\SymbolicUtils\2UXNG\src\rewriters.jl:144
 [14] (::SymbolicUtils.Rewriters.Fixpoint{SymbolicUtils.Rewriters.Walk{:post, SymbolicUtils.Rewriters.PassThrough{SymbolicUtils.Rewriters.RestartedChain{Vector{SymbolicUtils.AbstractRule}}}, SymbolicUtils.var"#simterm#125"{SymbolicUtils.var"#simterm#114#126"}, false}})(x::SymbolicUtils.Add{Number, Int64, Dict{Any, Number}, Nothing})
    @ SymbolicUtils.Rewriters C:\Users\asukys\.julia\packages\SymbolicUtils\2UXNG\src\rewriters.jl:104
 [15] to_mpoly(t::SymbolicUtils.Add{Number, Int64, Dict{Any, Number}, Nothing}, variable_type::Type, dicts::Tuple{OrderedCollections.OrderedDict{SymbolicUtils.Sym, Any}, OrderedCollections.OrderedDict{Any, SymbolicUtils.Sym}})
    @ SymbolicUtils C:\Users\asukys\.julia\packages\SymbolicUtils\2UXNG\src\abstractalgebra.jl:91
 [16] to_mpoly
    @ C:\Users\asukys\.julia\packages\SymbolicUtils\2UXNG\src\abstractalgebra.jl:78 [inlined]
 [17] expand(expr::SymbolicUtils.Add{Number, Int64, Dict{Any, Number}, Nothing}, variable_type::Type) (repeats 2 times)
    @ SymbolicUtils C:\Users\asukys\.julia\packages\SymbolicUtils\2UXNG\src\abstractalgebra.jl:173
 [18] top-level scope
    @ none:1
```
Seems to be related to DynamicPolynomials, maybe worth addressing @blegat?

**Comment by shashi** [comment]
It's fixed in #231 (merging it soon)

**Comment by shashi** [comment]
`expand` now converts back to SU types. I'm not sure why we're talking about VariableSource here.
